File: 121_best_time_to_buy_and_sell_stock/121_best_time_to_buy_and_sell_stock.c
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let results = 0
    let lowest = prices[0];

    for(let price of prices) {
       lowest = price > lowest? lowest : price
       results = results > price - lowest ? results: price - lowest 
    }
    return results
};

File: 121_best_time_to_buy_and_sell_stock/121_best_time_to_buy_and_sell_stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
  let l = 0;
  let r = 1;
  let profit = 0;
  while (r < prices.length) {
    if (prices[r] > prices[l]) {
      profit = Math.max(profit, prices[r] - prices[l]);
    } else l = r;
    r++;
  }
  return profit;
};
